test/Models/CsvCatalogItemSpec.js

```diff
@@ -94,31 +111,318 @@ describe('CsvCatalogItem', function() {
+        csvItem.url = 'test/csv/minimal.csv';
+        csvItem.load().then(function() {
+            expect(csvItem._tableDataSource).toBeDefined();
+            expect(csvItem._tableDataSource.dataset).toBeDefined();
+            expect(csvItem._tableDataSource.dataset.getRowCount()).toEqual(2);
+        }).otherwise(fail).then(done);
     });
 
-    it('is correctly loading csv data from text', function() {
-        expect(csvItem instanceof CatalogItem).toBe(true);
+    it('throws an error on non-csv file', function(done) {
+        csvItem.url = 'test/GeoJSON/polygon.topojson';
+        expect(true).toBe(true);
+        csvItem.load().then(fail).otherwise(done);
     });
 
-    it('returns an error on non-csv file', function() {
-        expect(csvItem instanceof CatalogItem).toBe(true);
+    it('identifies "lat" and "lon" fields', function(done) {
+        csvItem.updateFromJson( { data: 'lat,lon,value\n-37,145,10' });
+        csvItem.load().then(function() {
+            expect(csvItem._tableDataSource.dataset.hasLocationData()).toBe(true);
+        }).otherwise(fail).then(done);
+    });
+    it('identifies "latitude" and "longitude" fields', function(done) {
+        csvItem.updateFromJson( { data: 'latitude,longitude,value\n-37,145,10' });
+        csvItem.load().then(function() {
+            expect(csvItem._tableDataSource.dataset.hasLocationData()).toBe(true);
+        }).otherwise(fail).then(done);
+    });
+    it('does not mistakenly identify "latvian" and "lone_person" fields', function(done) {
+        csvItem.updateFromJson( { data: 'latvian,lone_person,lat,lon,value\n-37,145,-37,145,10' });
+        csvItem.load().then(function() {
+            expect(csvItem._tableDataSource.dataset.getVariableNamesByType(VarType.LON)).toEqual(['lon']);
+            expect(csvItem._tableDataSource.dataset.getVariableNamesByType(VarType.LAT)).toEqual(['lat']);
+        }).otherwise(fail).then(done);
+    });
+    it('handles numeric fields containing (quoted) thousands commas', function(done) {
+        csvItem.updateFromJson( { data: 'lat,lon,value\n-37,145,"1,000"\n-38,145,"234,567.89"' });
+        csvItem.load().then(function() {
+            expect(csvItem._tableDataSource.dataset.hasLocationData()).toBe(true);
+            expect(csvItem._tableDataSource.dataset.getDataValue('value', 0)).toEqual(1000);
+            expect(csvItem._tableDataSource.dataset.getDataValue('value', 1)).toBeCloseTo(234567.89,2);
+        }).otherwise(fail).then(done);
     });
 
-    it('returns an error on non-csv text', function() {
-        expect(csvItem instanceof CatalogItem).toBe(true);
+    it('matches LGAs by code', function(done) {
+        csvItem.updateFromJson( { data: 'lga_code,value\n31000,1' });
+        csvItem.load().then(function() {
+            expect(csvItem._regionMapped).toBe(true);
+            expect(csvItem.colorFunc).toBeDefined();
+            // 242 is the shapefile index of LGA boundary 31000. What a crappy way to test...
```

Comment:
Can you look at the created entities maybe?  `expect(csvItem._tableDataSource.czmlDataSource.entites.values[0].point.color.getValue()).not.toEqual(new Color(0,0,0,0))`.  Still pretty ugly but at least there's no magic 242 involved.